pkg/runtime/helper.go

```diff
@@ -46,6 +46,59 @@ func UnsafeObjectConvertor(scheme *Scheme) ObjectConvertor {
 	return unsafeObjectConvertor{scheme}
 }
 
+// PreferredGroupVersion returns the desired group and version reported by the GroupVersioner, or
+// false if no such group version was indicated.
+func PreferredGroupVersion(target GroupVersioner) (unversioned.GroupVersion, bool) {
+	if group, ok := target.PrefersGroup(); ok {
```

Comment:
The target says which group it prefers.  If it says that, we prefer that
group.

On Mon, Jun 6, 2016 at 8:21 AM, David Eads <notifications@github.com> wrote:

> In pkg/runtime/helper.go
> <https://github.com/kubernetes/kubernetes/pull/26044#discussion_r65880477>
> :
>
> > @@ -46,6 +46,59 @@ func UnsafeObjectConvertor(scheme *Scheme) ObjectConvertor {
> >  	return unsafeObjectConvertor{scheme}
> >  }
> >
> > +// PreferredGroupVersion returns the desired group and version reported by the GroupVersioner, or
> > +// false if no such group version was indicated.
> > +func PreferredGroupVersion(target GroupVersioner) (unversioned.GroupVersion, bool) {
> > +	if group, ok := target.PrefersGroup(); ok {
>
> This construct is reading weird for me. How are you choosing a group
> without the GroupKind?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26044/files/a16b294e553924a3213ad4119f03c3ba2953965e#r65880477>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p_r6VNqgb0szFDRXOnoZm-OO8Gs1ks5qJBDcgaJpZM4IkFHI>
> .
>